test_node.py

```diff
@@ -1,19 +1,133 @@
+from dataclasses import dataclass
+from os.path import basename
+
+# from typing import List
+
 import pytest
-import testinfra
+
+
+from conftest import restrict_to_version
+
 
 # Container fixture contains the black magic to run command on all the different kind of nodes
 # per language.
 # If you need to run a test for a single version, create your own fixture
 # You should think of reusing the `container` fixture from yours.
 
+
+@dataclass(frozen=True)
+class NpmPackageTest:
+    # node_versions: List[int] = field(default_factory=list)
+    build_command: str = ""
+    repository_url: str = ""
+
+    @property
+    def repo_name(self) -> str:
+        return basename(self.repository_url.replace(".git", ""))
+
+    @property
+    def test_command(self) -> str:
+        return f"""git clone {self.repository_url} &&
+cd {self.repo_name} &&
+{self.build_command}
+"""
+
+    def __str__(self) -> str:
+        return self.repo_name
+
+    def to_pytest_param(self, *args, **kwargs):
+        return pytest.param(self, id=self.__str__(), *args, **kwargs)
+
+
 # The container fixture automatically finds the file from the test, guesses the language, and starts all the necessary containers
 # See also conftest.py
 def test_node_version(container):
-    assert "v{}".format(container.version) in container.connection.check_output(
+    assert f"v{container.version}" in container.connection.check_output(
         "node -v"
     )
 
 
 # We don't care about the version, just test that the command seem to work
 def test_npm(container):
     assert container.connection.run_expect([0], "npm version")
+    assert container.connection.run_expect([0], "yarn --version")
+
+
+@pytest.mark.parametrize(
+    "npm_package",
+    [
+        pkg.to_pytest_param()
+        for pkg in (
+            NpmPackageTest(
+                repository_url="https://github.com/lodash/lodash.git",
+                build_command="npm ci && npm test",
+            ),
+            NpmPackageTest(
+                repository_url="https://github.com/chalk/chalk.git",
+                build_command="npm install && npm test",
+            ),
+            NpmPackageTest(
+                repository_url="https://github.com/tj/commander.js.git",
+                build_command="npm ci && npm test && npm run lint",
+            ),
+            NpmPackageTest(
+                repository_url="https://github.com/visionmedia/debug.git",
+                build_command="npm install && npm run lint && npm run test:node",
+            ),
+            NpmPackageTest(
+                repository_url="https://github.com/expressjs/express.git",
+                build_command="""npm config set shrinkwrap false &&
+            npm rm --silent --save-dev connect-redis &&
```

Comment:
Ok the only thing I find to say is that those lines are ugly, and would be better if they were all indented 4 spaces below build_command.

Reply:
Now better?